music_assistant/server/providers/siriusxm/__init__.py

```diff
@@ -203,7 +205,7 @@ async def get_stream_details(self, item_id: str) -> StreamDetails:
         """Get streamdetails for a track/radio."""
         hls_path = f"http://{self._base_url}/{item_id}.m3u8"
 
-        return StreamDetails(
+        self._stream_details = StreamDetails(
```

Comment:
I would rename this to something like _current_stream_details and add some comments what you are doing here as this is a bit fancy maybe for outsiders to understand.